package.json

```diff
@@ -37,10 +37,10 @@
     "pretest": "npm run-script autolint",
     "test": "cross-env NODE_ENV=test nyc mocha-webpack --opts ./mocha-webpack.opts --webpack-config ./webpack.config.test.js -r test/setup.js test",
     "codecov": "nyc report --reporter=lcov && codecov",
-    "clean": "git clean -fX ./dist ./*.xpi && npm run :clean:doc ",
+    "clean": "git clean -fX ./dist ./*.xpi ./addon.env && npm run :clean:doc ",
     "distclean": "git clean -dfX",
     "prepackage": "npm run-script clean && cross-env NODE_ENV=production npm run-script build",
-    "package": "cd dist && nodezip -c ../addon.xpi *"
+    "package": "echo 'TESTPILOT_ADDON_ID='${npm_package_id}'\nTESTPILOT_ADDON_VERSION='${npm_package_version} > addon.env && cd dist && nodezip -c ../addon.xpi *"
```

Comment:
Are the single quotes necessary? I'm pretty sure this will make Windows unhappy.

Reply:
>Are the single quotes necessary? I'm pretty sure this will make Windows unhappy.

Yes. Or at least: I wasn't able to find the right combination of output to both a) get the variables properly output and b) write to the addon.env file.

In other words, without single quotes around _just_ the text strings (sans variables):

```
"package": "echo TESTPILOT_ADDON_ID=${npm_package_id}\nTESTPILOT_ADDON_VERSION=${npm_package_version} > addon.env && cd dist && nodezip -c ../addon.xpi *"```

...nothing actually is written to addon.env in my environment.

And with single quotes around the entire echo:

```
"package": "echo 'TESTPILOT_ADDON_ID=${npm_package_id}\nTESTPILOT_ADDON_VERSION=${npm_package_version}' > addon.env && cd dist && nodezip -c ../addon.xpi *"
```

...the file is written to, but with "${npm_package_id}" and "${npm_package_version}" output  instead of the expected parsed variable ("lockbox@mozilla.com", "0.1.0-pre").

Therefore... I arrived at what you currently see here. 😬 

Any other ideas worth trying right now?

Reply:
I think the newline is causing issues. I would split it into two statements: `echo TESTPILOT_ADDON_ID=${npm_package_id} > addon.env && echo TESTPILOT_ADDON_VERSION=${npm_package_version} >> addon.env`

Reply:
I had some success with the following:

```sh
"package": "echo \"TESTPILOT_ADDON_ID=$npm_package_id\nTESTPILOT_ADDON_VERSION=$npm_package_version\" > addon.env && cd dist && nodezip -c ../addon.xpi *"
```

That said, this feels funny and dirty to me, and very Mac/UNIX specific and I'm not sure how well it'd perform on Windows. Not sure if it'd be better to just replace with some basic Node shell script:

```sh
#!/usr/bin/env node

const {writeFileSync} = require('fs');
const {id, version} = require('../package.json');

const output = `TESTPILOT_ADDON_ID=${id}\nTESTPILOT_ADDON_VERSION=${version}`;

writeFileSync('./addon.env', output);
```

Reply:
Good suggestions. Thanks, everyone.

I don't have a great way to test this until I actually push to production. So while I agree this feels a bit brittle and O/S specific, but to avoid more pollution or clutter here, I'll take the "good enough" route for now and make sure this builds in the Jenkins env leaving room to improve if it does actually work. 😉